# ICARUS v6.0 - Sistema de GestÃ£o OPME

## ğŸš€ Stack TecnolÃ³gica Oficial

### Frontend
| Tecnologia | VersÃ£o | Uso |
|------------|--------|-----|
| React | 18.3.1 | UI Framework |
| TypeScript | 5.9 | Type Safety |
| Vite | 6.4 | Build Tool |
| Tailwind CSS | 4.1 | Styling |
| Radix UI | Latest | Primitives |
| Framer Motion | 12.x | Animations |
| Recharts | 2.15 | Charts |
| Lucide React | Latest | Icons |

### Backend
| Tecnologia | VersÃ£o | Uso |
|------------|--------|-----|
| Supabase | Latest | BaaS |
| Edge Functions | Deno | Serverless |
| PostgreSQL | 16 | Database |
| pgvector + HNSW | Latest | Vector Search |
| Redis (Upstash) | Latest | Cache |

### InteligÃªncia Artificial
| Tecnologia | VersÃ£o | Uso |
|------------|--------|-----|
| LangChain | 0.3 | AI Framework |
| LangGraph | 0.2 | Agent Workflows |
| Claude 3.5 Sonnet | Latest | Primary LLM |
| GPT-4o | Latest | Secondary LLM |
| text-embedding-3-small | Latest | Embeddings |

### Mobile
| Tecnologia | VersÃ£o | Uso |
|------------|--------|-----|
| React Native | Latest | Mobile Framework |
| Expo SDK | 50 | Development Platform |

### SeguranÃ§a & Compliance
| Tecnologia | Uso |
|------------|-----|
| SHA-256 | Audit Log Hash |
| Proof of Work | Blockchain Integrity |
| RLS | Row Level Security |
| LGPD | Data Protection |
| ANVISA RDC 665 | OPME Traceability |

---

## ğŸ“ Estrutura do Projeto

```
icarus-v6-complete/
â”œâ”€â”€ sql/                          # Database Schema
â”‚   â”œâ”€â”€ 001-extensions.sql        # PostgreSQL 16 + pgcrypto + vector
â”‚   â”œâ”€â”€ 002-core.sql              # Empresas, UsuÃ¡rios, Perfis
â”‚   â”œâ”€â”€ 003-cadastros.sql         # Clientes, Hospitais, MÃ©dicos
â”‚   â”œâ”€â”€ 004-estoque.sql           # Produtos, Lotes, MovimentaÃ§Ãµes
â”‚   â”œâ”€â”€ 005-cirurgias.sql         # Cirurgias, Rastreabilidade OPME
â”‚   â”œâ”€â”€ 006-financeiro.sql        # Open Finance, Pluggy, VigilÃ¢ncia
â”‚   â”œâ”€â”€ 007-fiscal-ia.sql         # NF-e, InfoSimples, pgvector HNSW
â”‚   â”œâ”€â”€ 008-blockchain-lgpd.sql   # Audit Log Blockchain, LGPD
â”‚   â””â”€â”€ 009-rls-security.sql      # Row Level Security
â”œâ”€â”€ components/
â”‚   â””â”€â”€ ui/
â”‚       â”œâ”€â”€ primitives.tsx        # GlassCard, NeuButton, NeuInput, Icon3D
â”‚       â””â”€â”€ data-display.tsx      # KPICard, Badge, StatusIndicator, AlertCard
â”œâ”€â”€ lib/
â”‚   â”œâ”€â”€ index.ts                  # Hooks, Utils, Supabase, Redis
â”‚   â””â”€â”€ langchain.ts              # LangChain + LangGraph Configuration
â”œâ”€â”€ edge-functions/
â”‚   â””â”€â”€ all-functions.ts          # All Supabase Edge Functions
â”œâ”€â”€ mobile/
â”‚   â””â”€â”€ app.tsx                   # React Native + Expo App Complete
â”œâ”€â”€ config/
â”‚   â”œâ”€â”€ tailwind.css              # Dark Glass Medical Design System
â”‚   â””â”€â”€ project-config.ts         # Vite, TypeScript, ESLint, etc.
â””â”€â”€ README.md
```

---

## ğŸ¨ Design System: Dark Glass Medical

### Cores Principais
```css
/* Backgrounds */
--bg-primary: #050508;
--bg-secondary: #0a0a0f;
--bg-tertiary: #12121a;

/* Glass Effect */
--glass-bg: rgba(255, 255, 255, 0.03);
--glass-border: rgba(255, 255, 255, 0.08);
--glass-blur: 16px;

/* Accents */
--accent-primary: #3b82f6;    /* Blue */
--accent-secondary: #14b8a6;  /* Teal */

/* Status */
--status-success: #22c55e;
--status-warning: #f59e0b;
--status-danger: #ef4444;
```

### Neumorphic 3D Effects
```css
/* Outer Shadow */
box-shadow: 4px 4px 8px rgba(0,0,0,0.5), -4px -4px 8px rgba(255,255,255,0.03);

/* Inner Shadow */
box-shadow: inset 4px 4px 8px rgba(0,0,0,0.5), inset -4px -4px 8px rgba(255,255,255,0.03);

/* Glow */
box-shadow: 0 0 20px rgba(59, 130, 246, 0.3);
```

---

## ğŸ¤– Arquitetura de IA

```
â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”
â”‚                    LangGraph Orchestrator                    â”‚
â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤
â”‚  â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”  â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”  â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”         â”‚
â”‚  â”‚  Financial  â”‚  â”‚   Medical   â”‚  â”‚ Compliance  â”‚         â”‚
â”‚  â”‚   Agent     â”‚  â”‚   Agent     â”‚  â”‚   Agent     â”‚         â”‚
â”‚  â”‚ (LangChain) â”‚  â”‚ (LangChain) â”‚  â”‚ (LangChain) â”‚         â”‚
â”‚  â””â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”˜  â””â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”˜  â””â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”˜         â”‚
â”‚         â”‚                â”‚                â”‚                 â”‚
â”‚         â–¼                â–¼                â–¼                 â”‚
â”‚  â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”   â”‚
â”‚  â”‚                      LLMs                           â”‚   â”‚
â”‚  â”‚   Claude 3.5 Sonnet (Primary) â”‚ GPT-4o (Secondary)  â”‚   â”‚
â”‚  â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜   â”‚
â”‚                           â”‚                                 â”‚
â”‚         â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”              â”‚
â”‚         â–¼                 â–¼                 â–¼              â”‚
â”‚  â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”     â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”     â”Œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”        â”‚
â”‚  â”‚ pgvector  â”‚     â”‚   Redis   â”‚     â”‚ Supabase  â”‚        â”‚
â”‚  â”‚   HNSW    â”‚     â”‚   Cache   â”‚     â”‚    DB     â”‚        â”‚
â”‚  â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜     â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜     â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜        â”‚
â””â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”˜
```

### Workflow de Auditoria Financeira
```
START â†’ categorize â†’ detect_anomalies â†’ analyze_fees â†’ generate_suggestions â†’ generate_report â†’ END
```

---

## â›“ï¸ Blockchain Audit Log

### Estrutura do Bloco
```typescript
{
  block_index: number,
  previous_hash: string,      // SHA-256 do bloco anterior
  hash: string,               // SHA-256 deste bloco
  nonce: number,              // Proof of Work
  difficulty: 2,              // Zeros iniciais no hash
  empresa_id: UUID,
  tabela: string,
  registro_id: UUID,
  acao: 'INSERT' | 'UPDATE' | 'DELETE',
  dados_antes: JSONB,
  dados_depois: JSONB,
  criado_em: timestamp
}
```

### ValidaÃ§Ã£o
```sql
SELECT * FROM validate_blockchain('empresa-uuid');
-- Retorna: valid, total_blocks, invalid_block, error_message
```

---

## ğŸ“Š pgvector com HNSW

### Ãndice Otimizado
```sql
CREATE INDEX idx_ml_vectors_hnsw ON ml_vectors 
USING hnsw (embedding vector_cosine_ops)
WITH (m = 16, ef_construction = 64);
```

### Busca SemÃ¢ntica
```sql
SELECT * FROM search_vectors(
  query_embedding := '[0.1, 0.2, ...]'::vector,
  p_empresa_id := 'uuid',
  match_threshold := 0.7,
  match_count := 10
);
```

---

## ğŸ”Œ IntegraÃ§Ãµes

### Open Finance (Pluggy)
- ConexÃ£o bancÃ¡ria automatizada
- SincronizaÃ§Ã£o de transaÃ§Ãµes
- AnÃ¡lise de tarifas
- DDA (DÃ©bito Direto Autorizado)

### InfoSimples
- Consulta CNPJ/CPF
- ValidaÃ§Ã£o CRM
- Consulta ANVISA (produtos e empresas)
- EmissÃ£o de NF-e

### Webhooks
- Pluggy: `POST /api/pluggy-webhook`
- Supabase Realtime: TransaÃ§Ãµes, Cirurgias

---

## ğŸš€ InstalaÃ§Ã£o

```bash
# Clone
git clone https://github.com/newortho/icarus-v6.git
cd icarus-v6

# Instalar dependÃªncias
npm install

# Configurar ambiente
cp .env.example .env
# Editar .env com suas credenciais

# Iniciar Supabase local
npx supabase start

# Executar migrations
npx supabase migration up

# Gerar tipos
npx supabase gen types typescript --local > src/types/database.ts

# Iniciar desenvolvimento
npm run dev
```

---

## ğŸ“± Mobile App

```bash
cd mobile

# Instalar Expo CLI
npm install -g expo-cli

# Instalar dependÃªncias
npm install

# Iniciar
expo start
```

---

## ğŸ” VariÃ¡veis de Ambiente

```env
# Supabase
VITE_SUPABASE_URL=
VITE_SUPABASE_ANON_KEY=
SUPABASE_SERVICE_ROLE_KEY=

# AI
ANTHROPIC_API_KEY=
OPENAI_API_KEY=

# Open Finance
PLUGGY_CLIENT_ID=
PLUGGY_CLIENT_SECRET=

# InfoSimples
INFOSIMPLES_TOKEN=

# Redis
UPSTASH_REDIS_REST_URL=
UPSTASH_REDIS_REST_TOKEN=
```

---

## ğŸ“‹ Compliance

### ANVISA RDC 665/2022
- âœ… Rastreabilidade completa de OPME
- âœ… Registros imutÃ¡veis
- âœ… Hash SHA-256 por registro

### LGPD
- âœ… Consentimento documentado
- âœ… Direitos do titular (15 dias)
- âœ… AnonimizaÃ§Ã£o de dados
- âœ… Log de acesso a dados pessoais

### SeguranÃ§a
- âœ… Row Level Security (RLS)
- âœ… Blockchain Audit Log
- âœ… Proof of Work
- âœ… Criptografia de dados sensÃ­veis

---

## ğŸ§ª Testes

```bash
# Lint
npm run lint

# Build
npm run build

# Validar blockchain
SELECT * FROM validate_blockchain('empresa-uuid');
```

---

## ğŸ“„ LicenÃ§a

ProprietÃ¡rio - NEW ORTHO Â© 2024

---

## ğŸ‘¥ Contato

- **Empresa**: NEW ORTHO
- **Email**: dax@newortho.com.br
- **Sistema**: ICARUS v6.0
